sdcm/utils/raft/common.py

```diff
@@ -0,0 +1,39 @@
+import logging
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def check_raft_ready_on_nodes(nodes: list["BaseNode"]) -> bool:
+    LOGGER.debug("Check raft feature state on node")
+
+    nodes_raft_status = []
+    for node in nodes:
+        if raft_ready := node.raft.is_ready():
+            nodes_raft_status.append(raft_ready)
+            continue
+        LOGGER.error("Node %s has raft state: %s", node.name, node.raft.get_status())
+
+    if not all(nodes_raft_status):
+        raise Exception("Raft is not ready")
+    LOGGER.debug("Raft is ready!")
+    return True
+
+
+def check_raft_enabled_on_nodes(nodes: list["BaseNode"]) -> bool:
+    LOGGER.debug("Check raft is enabled on nodes")
+    raft_enabled_state = [node.raft.enabled for node in nodes]
+    LOGGER.debug("Raft feature state: %s", raft_enabled_state)
+    raft_enabled = all(raft_enabled_state)
+    raft_disabled = all(map(lambda x: not x, raft_enabled_state))
+    if not raft_enabled and not raft_disabled:
+        raise Exception("Raft is not enabled on all nodes")
+
+    LOGGER.debug("Raft feature: %s on all nodes", raft_enabled)
+    return raft_enabled
```

Comment:
This code is too obfuscated.
Please, look at this example:

```
LOGGER.debug("Check that raft is enabled on all the nodes")
if not all(raft_enabled_state := [node.raft.enabled for node in nodes]):
    raise RaftException("Raft is not enabled on all the nodes: %s" % raft_enabled_state)
LOGGER.debug("Raft is enabled on all the nodes")
return True
```

Reply:
the idea of code is detect situation, when node in cluster has different settings, example half of nodes have raft enabled in config, and other nodes have raft disabled. this was a reason why i check that all nodes has raft enabled or raft disabled.

In your solution, the situation when some have raft configured, and other haven't will not be covered.

Reply:
Ok, here is the code snippet that satisfies the requirement you mentioned:
```
LOGGER.debug("Check that raft is enabled on all the nodes")
if len({node.raft.enabled for node in nodes}) != 1:
    raise RaftException("Raft is not enabled on all the nodes")
LOGGER.debug("Raft is enabled on all the nodes")
return True
```